src/LiveComponent/tests/Integration/LiveComponentHydratorTest.php

```diff
@@ -324,6 +326,111 @@ public function onEntireEntityUpdated($oldValue)
             ;
         }];
 
+        yield 'DTO with Doctrine ArrayCollection with persisted entities: (de)hydration works correctly' => [function () {
+            $firstProduct = create(ProductFixtureEntity::class, ['id' => 1, 'name' => 'first'])->object();
+            \assert($firstProduct instanceof ProductFixtureEntity);
+
+            $secondProduct = create(ProductFixtureEntity::class, ['id' => 2, 'name' => 'second'])->object();
+            \assert($secondProduct instanceof ProductFixtureEntity);
+
+            $productList = new ArrayCollection([$firstProduct, $secondProduct]);
+            \assert($productList instanceof ArrayCollection);
+
+            $doctrineEntityDto = create(HoldsArrayCollectionAndEntity::class)->object();
```

Comment:
Just use:

```php
$doctrineEntityDto = new HoldsArrayCollectionAndEntity();
```

`create()` is a nice way to create entities. But because this is a simple object, for clarity, we can instantiate directly.

Reply:
Changed!